model/src/main/kotlin/ProjectAnalyzerResult.kt

```diff
@@ -46,21 +46,6 @@ data class ProjectAnalyzerResult(
     @JsonInclude(JsonInclude.Include.NON_EMPTY)
     val issues: List<OrtIssue> = emptyList()
 ) {
```

Comment:
This check helped finding bugs in the past, is there any similar check in the dependency graph builder?

Reply:
There is no such check yet. I think, it would be possible to add one that checks the packages added to the builder against the list of package identifiers referenced by the graph. But I would prefer to do this in a separate PR.

I am a bit on the fence regarding such runtime checks. Are they really necessary if you have good test coverage?

Reply:
The problem here is that the amount of different project setups that can be passed to the analyzer is virtually unlimited, so we cannot test for everything. Therefore it's good to have some safeguards in place that fail with a meaningful error message instead of failing in random other places where it might be difficult to figure out the root cause. But I'm fine with doing that in a separate PR.